### PR TITLE
dtspo242: Create TF module for Azure app gateway

### PR DESCRIPTION
**Pull request by mokainos**
https://tools.hmcts.net/jira/browse/DTSPO-242
Create TF module for Azure app gateway
[ ] Yes
[ x] No
```

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+  }
+}
+
+data "azurerm_monitor_diagnostic_categories" "diagnostic_categories" {
+  count = length(var.frontends) != 0 ? 1 : 0
+
+  resource_id = azurerm_application_gateway.ag[0].id
+}
+
+data "azurerm_log_analytics_workspace" "log_analytics" {
```
Comment (timja) [inline main.tf:161]: this should be a workspace ID and not looking up the workspace directly

File: provider.tf [hunk 2]
```diff
@@ -0,0 +1,3 @@
+provider "azurerm" {
```
Comment (timja) [inline provider.tf:1]: i don't think we need this?

File: secret.tf [hunk 3]
```diff
@@ -0,0 +1,14 @@
+data "azurerm_key_vault" "main" {
+  count = (length(var.frontends) != 0  && var.store_privateip == "true")? 1 : 0
```
Comment (timja) [inline secret.tf:2]: can we use a boolean for store_privateip ?

```suggestion
  count = (length(var.frontends) != 0  && var.store_privateip == "true") ? 1 : 0
```

File: secret.tf [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+  name                = var.vault_name
+  resource_group_name = "core-infra-${var.subscription}-rg"
+}
+
+resource "azurerm_key_vault_secret" "internal-fe-lb-ip" {
+  count = (length(var.frontends) != 0  && var.store_privateip == "true")? 1 : 0
```
Comment (timja) [inline secret.tf:9]:
```suggestion
  count = (length(var.frontends) != 0  && var.store_privateip == "true") ? 1 : 0
```

**Review by timja — COMMENTED** [review]
few minor comments

File: provider.tf (hunk 2 above)
Comment (mokainos) [reply provider.tf:1]: deleted the file

File: main.tf (hunk 1 above)
Comment (mokainos) [reply main.tf:161]: Will modify the root and child module accordingly

File: README.md [hunk 5]
```diff
@@ -1,2 +1,39 @@
...
+| Name | Description | Type | Default | Required |
+|------|-------------|------|---------|:--------:|
+| common\_tags | Common Tags | `map(string)` | n/a | yes |
+| destinations | List of IP addresses to direct traffic to | `list(string)` | n/a | yes |
+| env | environment, will be used in resource names and for looking up the vnet details | `any` | n/a | yes |
+| frontends | n/a | `list(any)` | n/a | yes |
+| location | location to deploy resources to | `any` | n/a | yes |
+| max\_capacity | Maximum capacity for autoscaling | `number` | `10` | no |
+| min\_capacity | Minimum capacity for autoscaling | `number` | `2` | no |
+| oms\_env | Name of the enviornment for log analytics workspace | `any` | n/a | yes |
```
Comment (timja) [inline README.md:27]: needs updating

File: secret.tf [hunk 6]
```diff
@@ -0,0 +1,14 @@
+data "azurerm_key_vault" "main" {
+  count = (length(var.frontends) != 0  && var.store_privateip == "true") ? 1 : 0
```
Comment (timja) [inline secret.tf:2]: is there a reason `store_privateip` isn't a boolean?

Comment (mokainos) [reply secret.tf]: changed the code.

**Review by timja — APPROVED** [review]

**Pull request merged**